wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubApiClient.java

```diff
@@ -162,15 +218,30 @@ public String[] getTokenScopes(String authenticationToken)
         });
   }
 
-  private HttpRequest buildGithubApiRequest(URI uri, String authenticationToken) {
-    return HttpRequest.newBuilder(uri)
-        .headers(
-            "Authorization",
-            "token " + authenticationToken,
-            "Accept",
-            "application/vnd.github.v3+json")
-        .timeout(DEFAULT_HTTP_TIMEOUT)
-        .build();
+  /**
+   * Builds and returns HttpRequest to acces the GitHub API.
+   *
+   * @param uri request uri
+   * @param authenticationToken authentication token, can be NULL as the GitHub can handle some
+   *     requests without authentication
+   * @return HttpRequest object
+   */
+  private HttpRequest buildGithubApiRequest(URI uri, @Nullable String authenticationToken) {
+    if (isNullOrEmpty(authenticationToken)) {
+      return HttpRequest.newBuilder(uri)
+          .headers("Accept", "application/vnd.github.v3+json")
+          .timeout(DEFAULT_HTTP_TIMEOUT)
+          .build();
+    } else {
+      return HttpRequest.newBuilder(uri)
+          .headers(
+              "Authorization",
+              "token " + authenticationToken,
+              "Accept",
+              "application/vnd.github.v3+json")
+          .timeout(DEFAULT_HTTP_TIMEOUT)
+          .build();
```

Comment:
This is a code duplication, why not to define the headers array out of the `return` statement?
```JAVA
String[] headers = {"Accept", "application/vnd.github.v3+json"};
if (!isNullOrEmpty(authenticationToken)) {
  headers[2] = "Authorization";
  headers[3] = "token " + authenticationToken;
}
return HttpRequest.newBuilder(uri)
        .headers(headers)
        .timeout(DEFAULT_HTTP_TIMEOUT)
        .build();
```

Reply:
I'm not sure it's a duplication.
There is only one if statement, which makes the code easy to read.

Reply:
This is a duplication because there are several identical code lines:
```JAVA
  private HttpRequest buildGithubApiRequest(URI uri, @Nullable String authenticationToken) {
    if (isNullOrEmpty(authenticationToken)) {
>     return HttpRequest.newBuilder(uri)
          .headers("Accept", "application/vnd.github.v3+json")
>         .timeout(DEFAULT_HTTP_TIMEOUT)
>         .build();
    } else {
>     return HttpRequest.newBuilder(uri)
          .headers(
              "Authorization",
              "token " + authenticationToken,
              "Accept",
              "application/vnd.github.v3+json")
>         .timeout(DEFAULT_HTTP_TIMEOUT)
>         .build();
        }
    }
```

Reply:
It's a question of taste and it should not block merging of this pull request.

Reply:
I don't block the merging, I just express my point.

Reply:
Or a little bit simpler with wrapping to list:
```java
private HttpRequest buildGithubApiRequest(URI uri, String authenticationToken) {
  List<String> headers = Arrays.asList("Accept", "application/vnd.github.v3+json");
  if (!isNullOrEmpty(authenticationToken)) {
    headers.addAll(Arrays.asList("Authorization", "token " + authenticationToken));
  }
  return HttpRequest.newBuilder(uri)
      .headers(headers.toArray(String[]::new))
      .timeout(DEFAULT_HTTP_TIMEOUT)
      .build();
}
```

Reply:
It's recommended to have "Accept: application/vnd.github.v3+json" header

Reply:
This looks much better. Let's go with that.

Reply:
It fails with `java.lang.UnsupportedOperationException` on
```
headers.addAll(Arrays.asList("Authorization", "token " + authenticationToken));

```

Reply:
> This is a code duplication, why not to define the headers array out of the `return` statement?
> 
> ```java
> String[] headers = {"Accept", "application/vnd.github.v3+json"};
> if (!isNullOrEmpty(authenticationToken)) {
>   headers[2] = "Authorization";

Throws ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
```

>   headers[3] = "token " + authenticationToken;
> }
> return HttpRequest.newBuilder(uri)
>         .headers(headers)
>         .timeout(DEFAULT_HTTP_TIMEOUT)
>         .build();
> ```

Reply:
> Throws ArrayIndexOutOfBoundsException

I haven't tested it but I think wrapping to a list should work:
```Java
List<String> headers = Arrays.asList("Accept", "application/vnd.github.v3+json");
if (!isNullOrEmpty(authenticationToken)) {
  headers.addAll(Arrays.asList("Authorization", "token " + authenticationToken));
}
```